### DISCUSSION TITLE
Change the FileUpload directory before upload happen

### DISCUSSION THREAD

**Discussion by jazzy1986** [filamentphp/filament#13587] [Help]
When is FileUpload upload the file? any lifecycle to change the directory before file actually upload?

**Comment by leandrocfe** [comment]
When you submit the form. Do you mean, `->directory()` ?
https://filamentphp.com/docs/3.x/forms/fields/file-upload#configuring-the-storage-disk-and-directory

**Reply by jazzy1986** [reply]
no, i need to change the directory before submit, for example. i set ->directory('abc') as default, then before the file upload (i found out it upload after beforevalidate cycle ) i want to set a running id (generate when user click save on form) like "bb/xxx-xxx-1/ to ->directory() so that the file can store in folder group by running id.
